ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionAnnouncement.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright contributors to Hyperledger Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.eth.transactions;
+
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.hyperledger.besu.plugin.data.TransactionType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class TransactionAnnouncement {
+  private final Hash hash;
+  private final Optional<TransactionType> type;
+  private final Optional<Integer> size;
+
+  public TransactionAnnouncement(final Hash hash) {
+    this.hash = hash;
+    this.type = Optional.empty();
+    this.size = Optional.empty();
+  }
+
+  public TransactionAnnouncement(final Transaction transaction) {
+    this(transaction.getHash(), transaction.getType(), transaction.calculateSize());
+  }
+
+  public TransactionAnnouncement(final Hash hash, final TransactionType type, final Integer size) {
+    this.hash = hash;
+    this.type = Optional.ofNullable(type);
+    this.size = Optional.ofNullable(size);
```

Comment:
Why allowing nulls here? 
`type` and `size` should never be null at this point, otherwise it means there message was malformed, and its processing should have stopped before.

Reply:
Hum, if the transaction is announced by a protocol version older than Eth/68, only the hash is announced

Reply:
if the transaction is announced by a protocol < Eth/68 then the constructor `public TransactionAnnouncement(final Hash hash)` is used if I am not wrong

Reply:
I understood your point now. I changed the constructors to not accept `null` and throw instead. Also added a unit test for that. Thanks again!